pyfolio/timeseries.py

```diff
@@ -985,129 +985,60 @@ def rolling_sharpe(returns, rolling_sharpe_window):
-
-    # create initial linear fit from beginning of timeseries thru warm_up_days
-    # or the specified 'cone_fit_end_date'
-    if cone_fit_end_date is None:
-        returns = input_rets[:warm_up_days]
-    else:
-        returns = input_rets[input_rets.index < cone_fit_end_date]
-
-    perf_ts = cum_returns(returns, 1)
-
-    X = list(range(0, perf_ts.size))
-    X = sm.add_constant(X)
-    sm.OLS(perf_ts, list(range(0, len(perf_ts))))
-    line_ols = sm.OLS(perf_ts.values, X).fit()
-    fit_line_ols_coef = line_ols.params[1]
-    fit_line_ols_inter = line_ols.params[0]
-
-    x_points = list(range(0, perf_ts.size))
-    x_points = np.array(x_points) * fit_line_ols_coef + fit_line_ols_inter
+    Parameters
+    ----------
+    is_returns : pd.Series
+        In-sample daily returns of the strategy, noncumulative.
+         - See full explanation in tears.create_full_tear_sheet.
+    num_days : int
+        Number of days to project the probability cone forward.
+    cone_std : int, float, or list of int/float
+        Number of standard devations to use in the boundaries of
+        the cone. If multiple values are passed, cone bounds will
+        be generated for each value.
+    starting_value : int or float
+        Starting value of the out of sample period.
+    num_samples : int
+        Number of num_days length samples to draw from the in-sample
+        daily returns.
 
-    perf_ts_r = pd.DataFrame(perf_ts)
-    perf_ts_r.columns = ['perf']
+    Returns
+    -------
+    pd.DataFrame
+        Contains upper and lower cone boundaries. Column names are
+        strings corresponding to the number of standard devations
+        above (positive) or below (negative) the projected mean
+        cumulative returns.
+    """
+    np.random.seed(100)
+    samples = np.empty((num_samples, num_days))
 
-    warm_up_std_pct = np.std(perf_ts.pct_change().dropna())
-    std_pct = warm_up_std_pct * np.sqrt(std_scale_factor)
+    for i in range(num_samples):
+        samples[i, :] = is_returns.sample(num_days, replace=True)
 
-    perf_ts_r['line'] = x_points
-    perf_ts_r['sd_up'] = perf_ts_r['line'] * (1 + num_stdev * std_pct)
-    perf_ts_r['sd_down'] = perf_ts_r['line'] * (1 - num_stdev * std_pct)
+    cum_samples = np.cumprod(1 + samples, axis=1) * starting_value
 
-    std_pct = warm_up_std_pct * np.sqrt(std_scale_factor)
+    cum_mean = cum_samples.mean(axis=0)
+    cum_std = cum_samples.std(axis=0)
 
-    last_backtest_day_index = returns.index[-1]
-    cone_end_rets = input_rets[input_rets.index > last_backtest_day_index]
-    new_cone_day_scale_factor = int(1)
-    oos_intercept_shift = perf_ts_r.perf[-1] - perf_ts_r.line[-1]
+    if isinstance(cone_std, (float, int)):
+        cone_std = [cone_std]
 
-    # make the cone for the out-of-sample/live papertrading period
-    for i in cone_end_rets.index:
-        returns = input_rets[:i]
-        perf_ts = cum_returns(returns, 1)
+    cone_bounds = pd.DataFrame()
```

Comment:
Already initializing columns and index and then using `.loc` below would make this a bit more readable.